include/nbl/builtin/hlsl/workgroup/fft.hlsl

```diff
@@ -18,43 +17,41 @@ namespace fft
 // ---------------------------------- Utils -----------------------------------------------
 
 template<typename SharedMemoryAccessor, typename Scalar>
-void exchangeValues(NBL_REF_ARG(complex_t<Scalar>) lo, NBL_REF_ARG(complex_t<Scalar>) hi, uint32_t threadID, uint32_t stride, NBL_REF_ARG(MemoryAdaptor<SharedMemoryAccessor>) sharedmemAdaptor)
+void exchangeValues(NBL_REF_ARG(complex_t<Scalar>) lo, NBL_REF_ARG(complex_t<Scalar>) hi, uint32_t threadID, uint32_t stride, NBL_REF_ARG(SharedMemoryAccessor) sharedmemAccessor)
 {
     const bool topHalf = bool(threadID & stride);
+    // Ternary won't take structs so we use this aux variable
     vector <Scalar, 2> toExchange = topHalf ? vector <Scalar, 2>(lo.real(), lo.imag()) : vector <Scalar, 2>(hi.real(), hi.imag());
-    shuffleXor<SharedMemoryAccessor, vector <Scalar, 2> >::__call(toExchange, stride, sharedmemAdaptor);
+    complex_t<Scalar> toExchangeComplex = {toExchange.x, toExchange.y};
+    shuffleXor<SharedMemoryAccessor, complex_t<Scalar> >::__call(toExchangeComplex, stride, sharedmemAccessor);
```

Comment:
but then you don't use the vector at all?

Btw I never knew that subgroup shuffle works on structs, seems fishy (even subgroup extended types only mentions vectors)

Reply:
subgroup shuffle cannot work on structs, I just got confirmation

Reply:
nvm, forgot its a workgroup shuffle, it can work on anything as long as you can decompose into 4 byte chunks for banking